include/rdma/fi_domain.h

```diff
@@ -139,6 +139,7 @@ struct fi_mr_dmabuf {
 	int		fd;
 	uint64_t	offset;
 	size_t		len;
+	void 		*base_addr;
```

Comment:
better to use `uint64_t`, otherwise you will have a warning

```
prov/efa/src/efa_mr.c:565:31: warning: pointer of type 'void *' used in arithmetic [-Wpointer-arith]
    mr_attr->dmabuf->base_addr + mr_attr->dmabuf->offset,
                               ^
prov/efa/src/efa_mr.c:565:4: warning: passing argument 4 of 'ibv_reg_dmabuf_mr' makes integer from pointer without a cast [-Wint-conversion]
    mr_attr->dmabuf->base_addr + mr_attr->dmabuf->offset,
    ^~~~~~~
```

Reply:
This is a user facing API structure.  void * aligns with the value that he user has.

Internally, pointers should be converted to integers by casting to (uintptr_t)